R/plot.R

```diff
@@ -440,21 +441,25 @@ plot_predictions <- function(data,
     intervals <- data.table::dcast(intervals, ... ~ boundary,
                                    value.var = "prediction")
 
-    plot <- plot +
-      ggdist::geom_lineribbon(
-        data = intervals,
-        aes(
-          ymin = lower, ymax = upper,
-          # We use the fill_ramp aesthetic for this instead of the default fill
-          # because we want to keep fill to be able to use it for other
-          # variables
-          fill_ramp = factor(range, levels = sort(unique(range), decreasing = TRUE))
-        ),
-        lwd = 0.4
-      ) +
-      ggdist::scale_fill_ramp_discrete(
-        name = "range"
-      )
+    # only plot ranges if there are ranges to plot
+    if (!all(range == 0)) {
```

Comment:
What situation is this addressing? It seems redundant with the test L438

Reply:
L438 checks whether there is *anything*, this is checking whether there are any ribbons to be drawn (i.e. actual ranges, not only the median)

Reply:
Yes, but as far as I can tell, `intervals` doesn't contain the median. Only the other quantiles. So it is in effect the same test. If I'm mistaken, can you add a test that failed with the previous behaviour but works now? To be sure we keep a correct behaviour in the future.